specification/applicationinsights/resource-manager/microsoft.insights/stable/2015-05-01/favorites_API.json

```diff
@@ -0,0 +1,403 @@
+            "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteAdd": {
+            "$ref": "./examples/FavoriteAdd.json"
+          }
+        }
+      },
+      "patch": {
+        "description": "Updates a favorite that has already been added to an Application Insights component.",
+        "operationId": "Favorite_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FavoriteIdParameter"
+          },
+          {
+            "name": "favoriteProperties",
+            "description": "Properties that need to be specified to update the existing favorite.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite definition updated.",
+            "schema": {
+                "$ref": "#/definitions/ApplicationInsightsComponentFavorite"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteUpdate.json"
+          }
+        }
+      },
+      "delete": {
+        "description": "Remove a favorite that is associated to an Application Insights component.",
+        "operationId": "Favorite_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/FavoriteIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The favorite has been successfully removed from the Application Insights component."
+          }
+        },
+        "x-ms-examples": {
+          "FavoriteList": {
+            "$ref": "./examples/FavoriteDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationInsightsComponentFavorite": {
+      "description": "Properties that define a favorite that is associated to an Application Insights component.",
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "The user-defined name of the favorite."
+        },
+        "Config": {
```

Comment:
only certain top level properties are allowed by ARM. All RP specific resource properties should go inside the properties envelope. Is this new API or something which already exists on the service side?

Reply:
This is not a new API, it already exists.

Reply:
I believe I understand this. The ARM API requires things to be structured this way (CDS service returns properties that must be encased within the 'ARM' top-level properties).

However, our API exists outside of this structure, and returns these top level structures today and does not return the top-level envelope that ARM provides. Does this mean we do not require the top-level ARM envelope?

For instance, I see that our 'Components' API includes in its "definitions" the top-level 'Resource' container, but I don't see it being used anywhere there. Is this the requirement then, to define it but not use it in this case?